dashboard/modules/job/job_agent.py

```diff
@@ -117,14 +119,31 @@ async def delete_job(self, req: Request) -> Response:
                 text=f"Job {job_or_submission_id} does not exist",
                 status=aiohttp.web.HTTPNotFound.status_code,
             )
-        if job.type is not JobType.SUBMISSION:
-            return Response(
-                text="Can only delete submission type jobs",
-                status=aiohttp.web.HTTPBadRequest.status_code,
-            )
-
         try:
-            deleted = await self.get_job_manager().delete_job(job.submission_id)
+            job_mgr = self.get_job_manager()
+            if job.type is JobType.DRIVER:
+                deleted = await job_mgr.delete_from_job_table(job.job_id)
+            elif job.type is JobType.SUBMISSION:
+                deleted = await job_mgr.delete_job(job.submission_id)
+                if deleted:
+                    job_infos = await self._gcs_aio_client.get_all_job_info(
+                        timeout=None
+                    )
+                    for job_id, job_item in job_infos.items():
+                        if job_item.config.ray_namespace.startswith(
+                            ray_constants.RAY_INTERNAL_NAMESPACE_PREFIX
+                        ):
+                            # Skip jobs in any _ray_internal_namespace
+                            continue
+                        metadata = job_item.config.metadata
+                        job_submission_id = metadata.get(JOB_ID_METADATA_KEY)
+                        if job_submission_id and job_submission_id == job.submission_id:
```

Comment:
can we just re-use the `find_job_by_ids` function?

Reply:
Due to the inability to guarantee a one-to-one correspondence between submission_id and job_id, it was not possible to reuse this interface in order to thoroughly clean all job_ids associated with a submission_id.